src/vs/editor/common/controller/oneCursor.ts

```diff
@@ -622,15 +632,55 @@ export class OneCursorOp {
 			validatedViewPosition = cursor.convertModelPositionToViewPosition(validatedPosition.lineNumber, validatedPosition.column);
 		}
 
+		return this.move(cursor, inSelectionMode, validatedViewPosition, eventSource, ctx);
+	}
+
+	public static move(cursor:OneCursor, inSelectionMode: boolean, to:editorCommon.IPosition | string, eventSource: string, ctx: IOneCursorOperationContext): boolean {
```

Comment:
please move argument decoding to `cursor.ts` to do it only once even if there are 1000 cursors.

Please also make it clear which one is cursorMove and which one is the old one. They are all called move now.